scripts/optimize-old-videos.ts

```diff
@@ -0,0 +1,55 @@
+import { join, basename, extname } from 'path'
+import { remove, readdir, move, stat } from 'fs-extra'
+import { CONFIG } from '../server/initializers/constants'
+import { getVideoFileResolution, transcode, getVideoFileBitrate, getVideoFileFPS } from '../server/helpers/ffmpeg-utils'
+import { getMaxBitrate } from '../shared/models/videos'
+import { VideoRedundancyModel } from '../server/models/redundancy/video-redundancy'
+import { VideoModel } from '../server/models/video/video'
+import { getUUIDFromFilename } from '../server/helpers/utils'
+
+run()
+  .then(() => process.exit(0))
+  .catch(err => {
+    console.error(err)
+    process.exit(-1)
+  })
+
+async function run () {
+  const files = await readdir(CONFIG.STORAGE.VIDEOS_DIR)
+  for (const file of files) {
+    const inputPath = join(CONFIG.STORAGE.VIDEOS_DIR, file)
+    const videoBitrate = await getVideoFileBitrate(inputPath)
+    const fps = await getVideoFileFPS(inputPath)
+    const { videoFileResolution, isPortraitMode } = await getVideoFileResolution(inputPath)
+    const uuid = getUUIDFromFilename(file)
+
+    const isLocalVideo = await VideoRedundancyModel.isLocalByVideoUUIDExists(uuid)
+    const isMaxBitrateExceeded = videoBitrate > getMaxBitrate(videoFileResolution, fps)
+    if (uuid && isLocalVideo && isMaxBitrateExceeded) {
+      await optimizeVideo(uuid, inputPath, videoFileResolution, isPortraitMode)
+    }
+  }
+  console.log('Finished optimizing videos')
+}
+
+async function optimizeVideo (uuid: string, inputPath: string, resolution: number, isPortraitMode: boolean) {
```

Comment:
You could use https://github.com/Chocobozzz/PeerTube/blob/develop/server/lib/video-transcoding.ts#L10

For example, add an optional parameter `videoInputPath?: string` and rename the function `optimizeVideoFile`